src/components/TipControl.vue

```diff
@@ -0,0 +1,205 @@
+
+<template>
+  <div
+    class="tip-url__wrapper"
+    :title="isTipped ? 'Total tips (you tipped too)' : 'Total tips (click to retip the same URL)'"
+  >
+    <div
+      v-if="show"
+      class="overlay"
+      @click="toggleTip(false)"
+    />
+    <div
+      class="tip__container_wrapper"
+      @:click.stop
+    >
+      <div
+        class="retip__content"
+        :class="[{ active: show }]"
+        @click="toggleTip(!show)"
+      >
+        <img
+          class="tip__icon"
+          :src="iconTip"
+        >
+        <ae-amount
+          :amount="tip.total_amount"
+          :round="2"
+        />
+        <fiat-value :amount="tip.total_amount.toString()" />
+      </div>
+      <div
+        v-if="show"
+        class="tip__container"
+      >
+        <loading :show-loading="showLoading" />
+        <div
+          v-show="error && !showLoading"
+          class="text-center mb-2"
+        >
+          An error occurred while sending tip
+        </div>
+        <div
+          v-if="!showLoading"
+          class="input-group"
+        >
+          <input
+            v-model="message"
+            type="text"
+            class="form-control tip__message"
+            placeholder="Add message"
+          >
+        </div>
+        <div v-if="!showLoading">
+          <ae-input-amount v-model="value" />
+          <ae-button
+            :disabled="!(isSendTipDataValid || isSendMessageDataValid)"
+            @click="submitAction()"
+          >
+            Tip
+          </ae-button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import iconTip from '../assets/iconTip.svg';
+import iconTipped from '../assets/iconTipped.svg';
+import aeternity from '../utils/aeternity';
+import Backend from '../utils/backend';
+import { EventBus } from '../utils/eventBus';
+import util, { USE_DEEP_LINKS } from '../utils/util';
+import AeAmount from './AeAmount.vue';
+import FiatValue from './FiatValue.vue';
+import AeInputAmount from './AeInputAmount.vue';
+import Loading from './Loading.vue';
+import AeButton from './AeButton.vue';
+import { wallet } from '../utils/walletSearch';
+
+export default {
+  name: 'TipControl',
+  components: {
+    Loading,
+    FiatValue,
+    AeInputAmount,
+    AeAmount,
+    AeButton,
+  },
+  props: {
+    tip: { type: Object, required: true },
+  },
+  data() {
+    return {
+      fiatValue: 0.00,
+      value: 0,
+      show: false,
+      showLoading: false,
+      error: true,
+      USE_DEEP_LINKS,
+      message: '',
+    };
+  },
+  computed: {
+    ...mapGetters(['account', 'balance', 'isLoggedIn', 'loading']),
+    eventPayload() {
+      return `${this.tip.id}:${this.show}`;
+    },
+    // deepLink() {
```

Comment:
if it is not used, better not leave "commented probably-features"

Reply:
The deeplink should be used. Maybe @davidyuk can take a look at the flow and see if it needs adjustment.

Reply:
so better tag it WIP